.github/workflows/build.yml

```diff
@@ -0,0 +1,34 @@
+name: Unit tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: restore
+        run: |
+          sudo apt-get install -y libgtest-dev
+      - name: build
+        run: .build/build
+      - name: unittest
+        run: .build/run-unittest
+
+  windows:
+    runs-on: windows-latest
+    # NB: see https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md for what's available in this image
```

Comment:
I don't like seeing links to the moving `main`/`master` branch for any reason, can you please choose a commit hash or tag that's permanent?

Reply:
The caveat here is that `runs-on: windows-latest` is a moving target, and that moving link with `main` kind of matches that to some effect.

Even if we'll fix it as `runs-on: windows-2022`, it will still change arbitrarily by GitHub, and, sadly, having that non-permanent link is what will describe it best.

Reply:
I get that, but my experience says that generally it's very likely that a link to `main` branch will entirely stop working in a year, for example (because repos get often restructured, so what has a certain filename/path now is likely to be renamed/relocated somewhere else entirely). So you get no content instead of at least some content (and with that, it's easier to find where the latest version is if you need it than if you get a Not Found page).

But OK, I try to keep an eye on whether the links still work once in a while, so I'll take care of this one too.

Reply:
> I get that, but my experience says that generally it's very likely that a link to `main` branch will stop working in a year

@GreyCat Just noting that I was right - https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md is now dead. It's actually dead for more than a year now, because it was moved in commit https://github.com/actions/runner-images/commit/d1f2c9a3bedac344533bd2499133f030adab7d90#diff-d3119a278ce9ce80fc4b5af63a1112fb3a044aec3e6002d838e8eb26d7a3a499 (authored on Nov 15, 2023) to `images/windows/Windows2022-Readme.md`.

Reply:
That's genuinely sad :(

I guess the only saving grace is that it's still a git repo and it's possible to get this file history tracked...

But I agree with you, modern software development front seem to rot faster than anything...